lineapy/api/api.py

```diff
@@ -140,51 +141,68 @@ def save(reference: object, name: str) -> LineaArtifact:
     return linea_artifact
 
 
-def delete(
-    artifact_name: str, version: Optional[Union[int, str]] = None
-) -> None:
+def delete(artifact_name: str, version: Union[int, str]) -> None:
     """
     Deletes an artifact from artifact store. If no other artifacts
     refer to the value, the value is also deleted from both the
     value node store and the pickle store.
 
-    If version is not provided, latest version is used.
+    :param artifact_name: Key used to while saving the artifact
+    :param version: version number or 'latest' or 'all'
+
+    :raises ValueError: if arifact not found or version invalid
     """
+    version = parse_artifact_version(version)
+
+    # get database instance
     execution_context = get_context()
     executor = execution_context.executor
     db = executor.db
 
-    get_version = None if not isinstance(version, int) else version
-    artifact = db.get_artifact_by_name(artifact_name, version=get_version)
+    # if version is 'all' or 'latest', get_version is None
+    get_version = None if isinstance(version, str) else version
+
+    try:
+        artifact = db.get_artifact_by_name(artifact_name, version=get_version)
+    except UserException:
+        raise NameError(
+            f"{artifact_name} not found. Perhaps there was a typo. Please try lineapy.artifact_store() to inspect all your artifacts."
+        )
 
     node_id = artifact.node_id
     execution_id = artifact.execution_id
 
-    num_artifacts = db.number_of_artifacts_per_node(node_id, execution_id)
-    if num_artifacts == 1:
+    db.delete_artifact_by_name(artifact_name, version=version)
+    logging.info(f"Deleted Artifact: {artifact_name} version: {version}")
+
+    try:
+        db.delete_node_value_from_db(node_id, execution_id)
+    except UserException:
+        logging.info(
+            f"Node: {node_id} with execution ID: {execution_id} not found in DB"
+        )
+
+    pickled_path = None
+    try:
+        pickled_path = db.get_node_value_path(node_id, execution_id)
+    except ValueError:
+        logging.info(f"No valid pickle path found for {node_id}")
```

Comment:
user does not need to know the node_id. this can be confusing so convert to logging.debug instead of info.